pkg/logs/internal/launchers/container/tailerfactory/file.go

```diff
@@ -0,0 +1,260 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+// +build docker
+
+package tailerfactory
+
+// This file handles creating docker tailers which access the container runtime
+// via files.
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	coreConfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/internal/launchers/container/tailerfactory/tailers"
+	"github.com/DataDog/datadog-agent/pkg/logs/internal/status"
+	"github.com/DataDog/datadog-agent/pkg/logs/internal/util/containersorpods"
+	"github.com/DataDog/datadog-agent/pkg/logs/sources"
+	dockerutilPkg "github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/filesystem"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+var podLogsBasePath = "/var/log/pods"
+var dockerLogsBasePath = "/var/lib/docker"
+var podmanLogsBasePath = "/var/lib/containers"
+
+// makeFileTailer makes a file-based tailer for the given source, or returns
+// an error if it cannot do so (e.g., due to permission errors)
+func (tf *factory) makeFileTailer(source *sources.LogSource) (Tailer, error) {
+	fileSource, err := tf.makeFileSource(source)
+	if err != nil {
+		return nil, err
+	}
+	return tf.attachChildSource(source, fileSource)
+}
+
+// makeFileSource makes a new LogSource with Config.Type=="file" to log the
+// given source.
+func (tf *factory) makeFileSource(source *sources.LogSource) (*sources.LogSource, error) {
+	// The user configuration consulted is different depending on what we are
+	// logging.  Note that we assume that by the time we have gotten a source
+	// from AD, it is clear what we are logging.  The `Wait` here should return
+	// quickly.
+	logWhat := tf.cop.Wait(context.Background())
+
+	switch logWhat {
+	case containersorpods.LogContainers:
+		// determine what the type of file source to create depending on the
+		// container runtime
+		switch source.Config.Type {
+		case "docker":
+			return tf.makeDockerFileSource(source)
+		default:
+			return nil, fmt.Errorf("file tailing is not supported for source type %s", source.Config.Type)
+		}
+
+	case containersorpods.LogPods:
+		return tf.makeK8sFileSource(source)
```

Comment:
I think the K8s code needs to be split out from the file code. 
Currently all k8s code is behind the `kubelet` build tag. It looks like windows builds include the `docker` build tag but NOT the `kubelet` tag which is causing unsupported tests to run on windows (hence the failures)